plugin/core/main.py

```diff
@@ -1,29 +1,122 @@
 import sublime
+import sublime_plugin
 
 from ..color import remove_color_boxes
 from ..diagnostics import DiagnosticsPresenter
 from ..highlights import remove_highlights
+from .handlers import LanguageHandler
 from .logging import set_debug_logging, set_exception_logging
 from .panels import destroy_output_panels, ensure_panel, PanelName
 from .popups import popups
-from .registry import windows, load_handlers, unload_sessions
+from .protocol import Response
+from .protocol import WorkspaceFolder
+from .registry import windows, unload_sessions
+from .rpc import method2attr
+from .sessions import AbstractPlugin
+from .sessions import register_plugin
+from .sessions import Session
+from .settings import client_configs
 from .settings import settings, load_settings, unload_settings
-from .typing import Optional
+from .transports import kill_all_subprocesses
+from .types import ClientConfig
+from .typing import Optional, List, Type, Callable, Dict
+import weakref
 
 
 def ensure_server_panel(window: sublime.Window) -> Optional[sublime.View]:
     return ensure_panel(window, PanelName.LanguageServers, "", "", "Packages/LSP/Syntaxes/ServerLog.sublime-syntax")
 
 
+def _get_final_subclasses(derived: List[Type], results: List[Type]) -> None:
+    """
+    This function should be removed: https://github.com/sublimelsp/LSP/issues/899
+    """
+    for d in derived:
+        d_subclasses = d.__subclasses__()
+        if len(d_subclasses) > 0:
+            _get_final_subclasses(d_subclasses, results)
+        else:
+            results.append(d)
+
+
+def _forcefully_register_plugins() -> None:
+    """
+    This function should be removed: https://github.com/sublimelsp/LSP/issues/899
+    """
+    plugin_classes = []  # type: List[Type[AbstractPlugin]]
+    _get_final_subclasses(AbstractPlugin.__subclasses__(), plugin_classes)
+    for plugin_class in plugin_classes:
+        register_plugin(plugin_class)
+    language_handler_classes = []  # type: List[Type[LanguageHandler]]
+    _get_final_subclasses(LanguageHandler.__subclasses__(), language_handler_classes)
+    for language_handler_class in language_handler_classes:
+        # Create an ephemeral plugin that stores an instance of the LanguageHandler as a class instance. Custom requests
+        # and notifications will work.
+        class LanguageHandlerTransition(AbstractPlugin):
+
+            handler = language_handler_class()
+
+            @classmethod
+            def name(cls) -> str:
+                return cls.handler.name  # type: ignore
+
+            @classmethod
+            def configuration(cls) -> sublime.Settings:
+                file_base_name = cls.name()
+                if file_base_name.startswith("lsp-"):
+                    file_base_name = "LSP-" + file_base_name[len("lsp-"):]
```

Comment:
Maybe utilize `cls.package_name` when exists instead. Kinda feels less hacky and error-prone than special-casing `LSP-` prefix.

Reply:
Actually that would be more error-prone, looking at first LSP-* example outside of sublime organization - https://packagecontrol.io/packages/LSP-tagml